### PR TITLE
fix bug for fill_empty_deterministic_ not support complex half

### LINKED ISSUE #133157: RuntimeError: "fill_empty_deterministic_" not implemented for 'ComplexHalf'
### 🐛 Describe the bug

RuntimeError: "fill_empty_deterministic_" not implemented for 'ComplexHalf'

### Error logs

RuntimeError: "fill_empty_deterministic_" not implemented for 'ComplexHalf'

### Minified repro

RuntimeError: "fill_empty_deterministic_" not implemented for 'ComplexHalf'

### Versions

torch 2.2.1

cc @ezyang @chauhang @penguinwu

### PR DESCRIPTION
**Pull request by wenqiny**
Fixes #133157

cc @jbschlosser

### CODE REVIEW

File: aten/src/ATen/native/TensorFactories.h [hunk 1]
```diff
@@ -106,7 +106,7 @@ inline Tensor& fill_empty_deterministic_(Tensor& tensor) {
     AT_DISPATCH_V2(
       tensor.scalar_type(), "fill_empty_deterministic_", AT_WRAP([&]() {
         tensor.fill_(std::numeric_limits<scalar_t>::quiet_NaN());
-    }), AT_EXPAND(AT_FLOATING_TYPES), AT_EXPAND(AT_COMPLEX_TYPES), AT_EXPAND(AT_FLOAT8_TYPES), kBFloat16, kHalf);
+    }), AT_EXPAND(AT_FLOATING_TYPES), AT_EXPAND(AT_COMPLEX_TYPES), AT_EXPAND(AT_FLOAT8_TYPES), kBFloat16, kHalf, kComplexHalf);
```
Comment (Skylion007) [inline aten/src/ATen/native/TensorFactories.h:109]: The complication here is I m not sure if kComplexHalf is supported on CPU...

**Comment by Skylion007** [comment]
@ywq880611 Please fix the lint

File: aten/src/ATen/native/TensorFactories.h (hunk 1 above)
Comment (ywq880611) [reply aten/src/ATen/native/TensorFactories.h:109]: image

I tried it in my local, but I didn't check the code details that how complex works on CPU.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**